thumbor_aws/loaders/s3_loader.py

```diff
@@ -18,41 +20,40 @@ def _get_bucket(url):
     bucket_path = "/".join(url_by_piece[1:])
     return bucket_name, bucket_path
 
-def _validate_bucket(context,bucket):
-    if not context.config.S3_ALLOWED_BUCKETS:
-        return True
-
-    for allowed in context.config.S3_ALLOWED_BUCKETS:
-        #s3 is case sensitive
-        if allowed == bucket:
-            return True
 
-    return False
+def _validate_bucket(context, bucket):
+    allowed_buckets = context.config.get('S3_ALLOWED_BUCKETS', default=None)
+    return not allowed_buckets or bucket in allowed_buckets
 
 
+@return_future
 def load(context, url, callback):
-    
     enable_http_loader = context.config.get('AWS_ENABLE_HTTP_LOADER', default=False)
 
     if enable_http_loader and 'http' in url:
-        return http_loader.load(context, url, callback)
-      
+        http_loader.load(context, url, callback)
```

Comment:
Shouldn't we call ``load_sync`` from ``http_loader`` directly rather than ``load``? That's what's done in the ``https_loader``. @masom any idea on that?